pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -598,6 +605,25 @@ func isSCCAvailable(c kubernetes.Interface) (bool, error) {
 	return isResourceAvailable(c, "securitycontextconstraints", "security.openshift.io", "v1")
 }
 
+func isMonitoringAvailable(c kubernetes.Interface) (bool, error) {
+	prometheusRuleResourceAvailable, err := isResourceAvailable(c, "customresourcedefinitions/prometheusrules.monitoring.coreos.com", "apiextensions.k8s.io", "v1")
+	if err != nil {
+		return false, errors.Wrap(err, "failed to check if prometheusRule resource is available")
+	}
+
+	serviceMonitorResourceAvailable, err := isResourceAvailable(c, "customresourcedefinitions/servicemonitors.monitoring.coreos.com", "apiextensions.k8s.io", "v1")
+	if err != nil {
+		return false, errors.Wrap(err, "failed to check if serviceMonitor resource is available")
+	}
+
+	if prometheusRuleResourceAvailable && serviceMonitorResourceAvailable {
+		return true, nil
+	}
+
+	log.Printf("will not deploy monitoring manifests: not all monitoring resources are available: %s: %v, %s, %v", monitoringv1.PrometheusRuleKind, prometheusRuleResourceAvailable, monitoringv1.ServiceMonitorsKind, serviceMonitorResourceAvailable)
+	return false, nil
+}
+
```

Comment:
Return false if there is an error but print the error we don't want CNAO to fail only if monitoring cannot be activated.

Reply:
I will change in the function calling this on the controller so that it will not fail and just log it.